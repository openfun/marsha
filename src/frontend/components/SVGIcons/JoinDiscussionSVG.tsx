import React from 'react';

import SVGIcon, { SvgProps } from '.';

interface JoinDiscussionSVGProps extends SvgProps {
  title: string;
}

export const JoinDiscussionSVG = ({
  baseColor,
  height,
  hoverColor,
  title,
  width,
}: JoinDiscussionSVGProps) => {
  return (
    <SVGIcon
      baseColor={baseColor}
      height={height}
      hoverColor={hoverColor}
      title={title}
      viewBox={'0 0 16 20'}
      width={width}
    >
      <g fill="none" fillRule="evenodd" stroke="none" strokeWidth="1">
        <g transform="translate(-378 -986)">
          <g transform="translate(50 691)">
            <path
              d="M19.916 15.333A6.665 6.665 0 0113.25 22a6.643 6.643 0 01-6.167-4.183l-2.525-6.342a.833.833 0 011.033-1.1l.659.217c.466.15.85.508 1.033.966l1.067 2.675a.411.411 0 00.383.267h.35V4.708a1.042 1.042 0 012.083 0v6.875c0 .234.184.417.417.417a.413.413 0 00.417-.417V3.042a1.042 1.042 0 012.083 0v8.541c0 .234.183.417.417.417a.413.413 0 00.416-.417V4.292a1.042 1.042 0 012.084 0v7.291c0 .234.183.417.416.417a.413.413 0 00.417-.417V6.792a1.042 1.042 0 012.083 0v8.541z"
              transform="translate(324 293)"
            ></path>
          </g>
        </g>
      </g>
    </SVGIcon>
  );
};
