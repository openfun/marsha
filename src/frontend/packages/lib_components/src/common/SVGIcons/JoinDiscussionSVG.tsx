import React from 'react';

import SVGIcon, { SvgProps } from '.';

export const JoinDiscussionSVG = (svgProps: SvgProps) => {
  return (
    <SVGIcon viewBox={{ x: 0, y: 0, width: 54, height: 54 }} {...svgProps}>
      <g fillRule="evenodd" stroke="none" strokeWidth="1">
        <g transform="translate(0.5 2.5)">
          <path
            fillRule="nonzero"
            d="M41.492 31.944c0 7.674-6.215 13.89-13.889 13.89a13.84 13.84 0 01-12.847-8.716l-5.26-13.212c-.538-1.371.746-2.743 2.153-2.291l1.371.451a3.484 3.484 0 012.153 2.014l2.222 5.573a.856.856 0 00.799.555h.729V9.81a2.17 2.17 0 014.34 0v14.323a.86.86 0 00.868.868.86.86 0 00.868-.868V6.337a2.17 2.17 0 014.34 0v17.795a.86.86 0 00.869.868.86.86 0 00.868-.868V8.94a2.17 2.17 0 014.34 0v15.19a.86.86 0 00.868.869.86.86 0 00.868-.868v-9.983a2.17 2.17 0 014.34 0v17.795z"
          ></path>
        </g>
      </g>
    </SVGIcon>
  );
};
