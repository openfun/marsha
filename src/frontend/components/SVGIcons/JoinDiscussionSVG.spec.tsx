import React from 'react';
import renderer from 'react-test-renderer';
import { JoinDiscussionSVG } from './JoinDiscussionSVG';

it('renders JoinDiscussionSVG correctly', () => {
  const JoinDiscussionSVGSnapshot = renderer
    .create(<JoinDiscussionSVG iconColor={'#035ccd'} />)
    .toJSON();
  expect(JoinDiscussionSVGSnapshot).toMatchSnapshot();
});
