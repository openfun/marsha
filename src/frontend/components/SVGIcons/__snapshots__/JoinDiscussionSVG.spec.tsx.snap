// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders JoinDiscussionSVG correctly 1`] = `
.c0 path {
  fill: #035ccd;
}

<svg
  className="c0"
  viewBox="0 0 54 54"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(2.5 2.5)"
    >
      <path
        d="M41.492 31.944c0 7.674-6.215 13.89-13.889 13.89a13.84 13.84 0 01-12.847-8.716l-5.26-13.212c-.538-1.371.746-2.743 2.153-2.291l1.371.451a3.484 3.484 0 012.153 2.014l2.222 5.573a.856.856 0 00.799.555h.729V9.81a2.17 2.17 0 014.34 0v14.323a.86.86 0 00.868.868.86.86 0 00.868-.868V6.337a2.17 2.17 0 014.34 0v17.795a.86.86 0 00.869.868.86.86 0 00.868-.868V8.94a2.17 2.17 0 014.34 0v15.19a.86.86 0 00.868.869.86.86 0 00.868-.868v-9.983a2.17 2.17 0 014.34 0v17.795z"
        fillRule="nonzero"
      />
    </g>
  </g>
</svg>
`;
